crates/orchestrator/src/queue/job_queue.rs

```diff
@@ -20,20 +20,20 @@ pub struct JobQueueMessage {
     pub(crate) id: Uuid,
 }
 
-pub async fn add_job_to_process_queue(id: Uuid) -> Result<()> {
+pub async fn add_job_to_process_queue(id: Uuid) -> EyreResult<()> {
     log::info!("Adding job with id {:?} to processing queue", id);
     add_job_to_queue(id, JOB_PROCESSING_QUEUE.to_string(), None).await
 }
 
-pub async fn add_job_to_verification_queue(id: Uuid, delay: Duration) -> Result<()> {
+pub async fn add_job_to_verification_queue(id: Uuid, delay: Duration) -> EyreResult<()> {
     log::info!("Adding job with id {:?} to verification queue", id);
     add_job_to_queue(id, JOB_VERIFICATION_QUEUE.to_string(), Some(delay)).await
 }
 
-pub async fn consume_job_from_queue<F, Fut>(queue: String, handler: F) -> Result<()>
+pub async fn consume_job_from_queue<F, Fut>(queue: String, handler: F) -> EyreResult<()>
```

Comment:
should we add custom errors here as well? `FailedToConsumeFromQueue`, `FailedToHandleJob`

Reply:
```rust
pub async fn init_consumers() -> Result<(), JobError> {
    // TODO: figure out a way to generalize this
    tokio::spawn(async move {
        loop {
            match consume_job_from_queue(JOB_PROCESSING_QUEUE.to_string(), process_job).await {
                Ok(_) => {}
                Err(e) => log::error!("Failed to consume from queue {:?}. Error: {:?}", JOB_PROCESSING_QUEUE, e),
            }
            sleep(Duration::from_secs(1)).await;
        }
    });
```
### My thought process :
I checked the usage of `consume_job_from_queue` and it's used in `init_consumers`,
while matching the error we are only logging the error and moving ahead.
This code indirectly suggested that throwing custom error for `consume_job_from_queue` is not as important as others as it is no where stopping the code.

Though this is not a fact based decision, do you see any perks in creating custom errors here ? 
@apoorvsadana

Reply:
But the logging part is mostly true for other errors as well right? I think we're logging it in almost every case for now but as `lib`, it makes it generalised so that others could handle it if they like. One example is in test cases, we check for `QueueError` to have `NoData`

Reply:
```rust
 match job_message {
        Some(job_message) => {
            log::info!("Handling job with id {:?} for queue {:?}", job_message.id, queue);
            match handler(job_message.id).await {
                Ok(_) => delivery.ack().await.map_err(|(e, _)| e)?,
                Err(e) => {
                    log::error!("Failed to handle job with id {:?}. Error: {:?}", job_message.id, e);

                    // if the queue as a retry logic at the source, it will be attempted
                    // after the nack
                    delivery.nack().await.map_err(|(e, _)| e)?;
                }
            };
        }
        None => return Ok(()),
    };
```

Valid, in the above code the `handler` is the general `process_job` function, which is also being logged.
Makes sense, adding the Errors.

Reply:
Added the needed Error cases 
Please check @apoorvsadana.